frontend/public/components/volumes-table.tsx

```diff
@@ -52,39 +55,45 @@ const ContainerLink: React.FC<ContainerLinkProps> = ({name, pod}) => <span class
 </span>;
 ContainerLink.displayName = 'ContainerLink';
 
-const VolumeRow: React.FC<VolumeRowProps> = ({value, pod}) => {
-  const kind = _.get(getVolumeType(value.volume), 'id', '');
-  const loc = getVolumeLocation(value.volume);
-  const name = value.name;
-  const permission = value.readOnly ? 'Read-only' : 'Read/Write';
+const VolumeRow: React.FC<VolumeRowProps> = ({volume, resource}) => {
+  const type = _.get(getVolumeType(volume.volume), 'id', '');
+  const loc = getVolumeLocation(volume.volume);
+  const name = volume.name;
+  const permission = volume.readOnly ? 'Read-only' : 'Read/Write';
+  const pod = resource.kind === 'Pod'? resource : resource.spec.template;
 
   return <div>
-    {value.mounts.map((m: Mount, i: number) => <React.Fragment key={i}>
+    {volume.mounts.map((m: Mount, i: number) => <React.Fragment key={i}>
       <div className="row">
         <div className="col-lg-2 col-md-3 col-sm-4 col-xs-5">{name}</div>
         <div className="col-lg-2 col-md-3 col-sm-5 col-xs-7 co-break-all co-select-to-copy">{_.get(m, 'mountPath', '-')} </div>
         <div className={classNames('col-lg-2 col-md-2 col-sm-3 hidden-xs co-break-all', { 'co-select-to-copy': _.get(m, 'subPath') })}>
           {_.get(m, 'subPath', '-')}
         </div>
         <div className="col-lg-2 col-md-2 hidden-sm hidden-xs">
-          <VolumeIcon kind={kind} />
+          <VolumeIcon kind={type} />
           <span className="co-break-word">{loc && ` (${loc})`}</span>
         </div>
         <div className="col-lg-2 col-md-2 hidden-sm hidden-xs">{permission}</div>
-        <div className="col-lg-2 hidden-md hidden-sm hidden-xs">
+        <div className="col-lg-1 hidden-md hidden-sm hidden-xs">
           { _.get(pod, 'kind') === 'Pod'
             ? <ContainerLink name={m.container} pod={pod as PodKind} />
             : <div>{m.container}</div>
           }
         </div>
+        <div className="col-lg-1 hidden-md hidden-sm hidden-xs">
```

Comment:
We're hiding the action menu at smaller screen sizes?

Reply:
We previously had a way of adding the kebab without using a column in the bootstrap grid for list tables IIRC. We should try to follow the same pattern.

We might want to bit the bullet and switch to PF4 tables while we're changing this.

Reply:
Changed to pf tables for the VolumesTable component.